benchmarks/base_classes.py

```diff
@@ -344,3 +347,72 @@ class T2IAdapterSDXLBenchmark(T2IAdapterBenchmark):
 
     def __init__(self, args):
         super().__init__(args)
+
+
+class BaseBenchmarkTestCase:
+    model_class = None
+    pretrained_model_name_or_path = None
+    model_class_name = None
+
+    def __init__(self):
+        super().__init__()
+
+    def get_result_filepath(self, suffix):
+        name = (
+            self.model_class_name
+            + "_"
+            + self.pretrained_model_name_or_path.replace("/", "_")
+            + "_"
+            + f"{suffix}.csv"
+        )
+        filepath = os.path.join(BASE_PATH, name)
+        return filepath
+
+
+class AutoencoderKLBenchmark(BaseBenchmarkTestCase):
+    model_class = AutoencoderKL
+
+    def __init__(self, pretrained_model_name_or_path, dtype, **kwargs):
+        super().__init__()
+        self.dtype = getattr(torch, dtype)
+        model = self.model_class.from_pretrained(pretrained_model_name_or_path, torch_dtype=self.dtype, **kwargs).eval()
```

Comment:
Likewise, we could move all of these reusable components to `BaseBenchmarkTestCase` (perhaps rename to `ModelBaseBenchmarkTestCase`) and let the users specify `pretrained_model_name_or_path`, `torch_dtype`, `subfolder`, etc.